test/breaker.test.js

```diff
@@ -0,0 +1,127 @@
+describe('Test breakers', function () {
+  var MockRedisClient = require('./mockRedisClient.js');
+  var proxyquire = require('proxyquire');
+
+  // Reset before each test
+  var ShardedRedis;
+
+  /********************************
+   * Setup
+   ********************************/
+
+  beforeEach(function () {
+    // Get sharded-redis-client with the redis clients stubbed out
+    ShardedRedis = proxyquire('../sharded-redis-client.js', {
+      redis: MockRedisClient
+    });
+  });
+
+  /********************************
+   * Tests
+   ********************************/
+
+  it('should call the appropriate redis function with a closed breaker', function (done) {
+    spyOn(MockRedisClient.prototype, 'get').and.callThrough();
+    var numMasterHosts = 1;
+    var options = {
+      breakerConfig: {
+        failure_rate: 0.5,
+        failure_count: 10,
+        reset_timeout: 30000
+      }
+    };
+    var key = 'key';
+    var redisHosts = generateRedisHosts(numMasterHosts);
+
+    var shardedClient = new ShardedRedis(redisHosts, options);
+    var mockedClient = shardedClient._getWrappedClient(key).get();
+    spyOn(mockedClient._breaker, 'pass').and.callThrough();
+
+    shardedClient.get(key, function (err) {
```

Comment:
An idea for an improvement on this (type of) test - right now it looks like it's testing the internal workings of the sharded Redis client by checking whether certain methods are called. That might make the most sense here, but I'd also consider if there's a more API-based way to test for this functionality.

In other words, how does using a breaker change the behavior of the sharded Redis client as it would impact a consumer?

We'd probably want to test for that, because that's the contract the redis client provides to consumers and if a later refactoring breaks that contract, we want to be alerted to it through a failing unit test. However, if a developer refactors this `.get()` method without breaking the contract, do we really care?